sdk/storage/faq.md

```diff
@@ -0,0 +1,410 @@
+The blob created with this method doesn't have any blocks, which means functions like `StageBlock()`, `CommitBlockList()` or `GetBlockList()` don't apply here.
+You cannot append blocks to such blobs (You'll have to overwrite it with a new one).
+You want to use this one if you need precise control over the SDK behavior at an HTTP level.
+
+`BlockBlobClient::UploadFrom` takes a memory buffer or file name as a parameter, splits the data in the buffer or file into smaller chunks intelligently or according to some parameters if provided,
+and then upload the chunks with multiple threads.
+This one is suitable in most cases. You can expect higher throughput because the chunks are transferred concurrently. It's especially recommended if you need to transfer large blobs efficiently.
+
+## How to efficiently upload large amount of small blobs?
+
+Unfortunately, this SDK doesn't provide a convenient way to upload many blobs or directory contents (files and sub-directories) with just one function call.
+You have to create multiple threads, traverse the directories by yourself and upload blobs one by one in each thread to speed up the transfer. Below is a skeleton example.
+```C++
+const std::vector<std::string> paths; // Files to be uploaded
+std::atomic<size_t> curr{0};
+auto upload_func = [&]() {
+  while (true)
+  {
+    auto id = curr.fetch_add(1);
+    if (id >= paths.size())
+    {
+      break;
+    }
+    const std::string path = paths[id];
+    try
+    {
+      blobClient.UploadFrom(path);
+    }
+    catch (Azure::Storage::StorageException& e)
+    {
+      // exception handling
+    }
+  }
+};
+
+std::vector<std::thread> thread_handles;
+for (size_t i = 0; i < num_threads; ++i)
+{
+  thread_handles.push_back(std::thread(upload_func));
+}
+for (auto& i : thread_handles)
+{
+  i.join();
+}
+```
+Or you can use tools like [AzCopy](https://learn.microsoft.com/azure/storage/common/storage-ref-azcopy) or [Data Movement Library](https://learn.microsoft.com/azure/storage/common/storage-use-data-movement-library).
+
+## How to ensure data integrity with transactional checksum?
+
+Generally speaking, the TLS protocol includes checksum that's strong enough to detect accidental corruption or deliberate tampering.
+Another layer of checksum is usually not necessary if you're using HTTPS.
+
+If you really want it for whatever reason, for example, to detect corruptions before the data is written to the socket,
+you can leverage the transactional checksum feature in the SDK.
+With this feature, you provide a pre-calculated MD5 or CRC64 checksum when calling an upload API
+and the storage service will calculate checksum after it receives the data. It will then compare that with the one you provide
+and fail the request if they don't match.
+Make sure you calculate the checksum early enough to cover the time when the corruption may happen.
```

Comment:
```suggestion
Make sure you calculate the checksum as early as possible so that potential corruptions that occur on storage medium, computer memory or during data transfer can be detected.
```